File: RestaurantReviewsAPI/RestaurantReviewsUI/ClientApp/src/app/shared/models/review.ts
import { Restaurant } from './restaurant';
import { User } from './user';

export class Review {
  Id: number;
  Restaurant: Restaurant;
  User: User;
  Stars: number;
  Description: string;

  constructor(values: Object = {}) {
    (<any>Object).assign(this, values);
  }
}

File: RestaurantReviewsAPI/RestaurantReviewsAPI/Models/Review.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Security.Principal;
using System.Web;

namespace RestaurantReviewsAPI.Models
{
    public class Review : Entity
    {
        [Required]
        public int Stars { get; set; }
        [Required]
        public string Description { get; set; }

        // Foreign Key
        public int RestaurantId { get; set; }
        // Navigation property
        [Required]
        public Restaurant Restaurant { get; set; }

        public string UserId { get; set; }
        public ApplicationUser User { get; set; }
    }
}